classes/ActionScheduler_wpPostStore.php

```diff
@@ -638,4 +749,83 @@ public function init() {
 		$taxonomy_registrar = new ActionScheduler_wpPostStore_TaxonomyRegistrar();
 		$taxonomy_registrar->register();
 	}
+
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in GMT.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime
+	 * @throws InvalidArgumentException When the action cannot be found.
+	 */
+	public function get_last_attempt( $action_id ) {
+		$post = $this->get_valid_post_object( $action_id );
+
+		return as_get_datetime_object( $post->post_modified_gmt );
+	}
+
+	/**
+	 * Get the last time the action was attempted.
+	 *
+	 * The time should be given in the local time of the site.
+	 *
+	 * @param string $action_id
+	 *
+	 * @return DateTime
+	 * @throws InvalidArgumentException When the action cannot be found.
+	 */
+	public function get_last_attempt_local( $action_id ) {
+		$post = $this->get_valid_post_object( $action_id );
+
+		return as_get_datetime_object( $post->post_modified, get_option( 'gmt_offset', '0' ) );
+	}
+
+	/**
+	 * Get a valid WP_Post object based on the action ID.
+	 *
+	 * @author Jeremy Pry
+	 *
+	 * @param string $action_id
+	 *
+	 * @return WP_Post
+	 * @throws InvalidArgumentException When the action cannot be found.
+	 */
+	protected function get_valid_post_object( $action_id ) {
+		$post = get_post( $action_id );
+		if ( empty( $post ) || ( $post->post_type !== self::POST_TYPE ) ) {
+			throw new InvalidArgumentException( sprintf( __( 'Unidentified action %s', 'action-scheduler' ), $action_id ) );
+		}
+
+		return $post;
+	}
+
+	/**
+	 * Set the last attempt for the given action.
+	 *
+	 * @param string   $action_id The action ID to update.
+	 * @param DateTime $date The DateTime object representing the last attempt. If not provided, the current
+	 *                       time will be used for the last attempt.
+	 *
+	 * @return bool Whether setting the last attempt was successful.
+	 */
+	public function set_last_attempt_date( $action_id, DateTime $date = null ) {
+		if ( null === $date ) {
+			$date = as_get_datetime_object();
+		}
```

Comment:
Why is `$date` optional here when it's required on the definition in `ActionScheduler_Store`?

Reply:
I meant to go back and update the abstract definition, but missed that. I'll include that change.